### DISCUSSION TITLE
How to enable CORS on Vite app `public` folder (not Next.js)

### DISCUSSION THREAD

**Discussion by qodesmith** [vercel/community#879] [Help] [answered]
I'm not using Nextx.js, but am using Vite to deploy a basic React + TS app. I'd like to enable CORS on files in the `public` folder. For example, I have a bunch of mp3's I'd like to enable CORS on. I've failed with this `vercel.json` config:

```json
{
  "headers": [
    {
      "source": "/public/*.mp3",
      "headers": [{"key": "Access-Control-Allow-Origin", "value": "self"}]
    }
  ]
}
```

I've tried other variations of the `source` value with no luck, such as:
* `"*.mp3"`
* `"/*.mp3"`
* `"/**/*.mp3"`

Perhaps there's a way to just enable CORS on the entire `public` folder?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @qodesmith! You can [enable CORS using `vercel.json`](https://vercel.com/guides/how-to-enable-cors#enabling-cors-using-vercel.json) for Serverless Functions, as you must have already seen. I'm not sure about adding these headers to the public folder though. Can you explain more about what you're trying to do and what problems you're facing? I'm happy to help you come up with an appropriate solution.

**Reply by qodesmith** [reply]
Hey, thanks for the reply. I'm building a music website and I have a bunch of mp3's in the `public` folder. I guess I'm just trying to prevent anyone else from hot-linking to the files. It seems CORS would solve that - the music should only be playable from my website's domain.

**Reply by amyegan** [reply]
In that case you should move your mp3 files out of the [`public` directory](https://create-react-app.dev/docs/using-the-public-folder/) and into `src`. That will give you more control over how the files are used.

Depending on the size and number of files, and on how your site will be used, you might also consider an external hosting solution for the files. Things like [Amazon S3](https://aws.amazon.com/premiumsupport/knowledge-center/secure-s3-resources/), [Anchor](https://anchor.fm/), and [SoundCloud](https://soundcloud.com/) are popular ways to serve audio without impacting the rest of your site.
